### PR TITLE
gsagh-248 add editor config + clean up code

### PR DESCRIPTION
**Pull request by DominikaLos**

### CODE REVIEW

File: GsaGH/Components/0_Model/CreateModel.cs [hunk 1]
```diff
@@ -1,8 +1,10 @@
-﻿using System;
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
```
Comment (kpne) [inline GsaGH/Components/0_Model/CreateModel.cs:3]: Do we need need this license for each file?

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+csharp_space_between_method_call_name_and_opening_parenthesis = false
+csharp_space_between_method_call_parameter_list_parentheses = false
+csharp_space_between_method_declaration_empty_parameter_list_parentheses = false
+csharp_space_between_method_declaration_name_and_open_parenthesis = false
+csharp_space_between_method_declaration_parameter_list_parentheses = false
+csharp_space_between_parentheses = false
+csharp_space_between_square_brackets = false
+
+# License header
+file_header_template = Licensed to the .NET Foundation under one or more agreements.\nThe .NET Foundation licenses this file to you under the MIT license.
```
Comment (kpne) [inline .editorconfig:188]: @tlmnrnhrdt

File: GsaGH/Components/0_Model/CreateModel.cs (hunk 1 above)
Comment (DominikaLos) [reply GsaGH/Components/0_Model/CreateModel.cs:3]: deleted

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+
+# name all constant fields using PascalCase
+dotnet_naming_rule.constant_fields_should_be_pascal_case.severity = error
+dotnet_naming_rule.constant_fields_should_be_pascal_case.symbols  = constant_fields
+dotnet_naming_rule.constant_fields_should_be_pascal_case.style = pascal_case_style
+dotnet_naming_symbols.constant_fields.applicable_kinds   = field
+dotnet_naming_symbols.constant_fields.required_modifiers = const
+dotnet_naming_style.pascal_case_style.capitalization = pascal_case
+
+# static fields should have s_ prefix
```
Comment (tlmnrnhrdt) [inline .editorconfig:95]: @kpne @DominikaLos can we get rid off the s_ prefix pls?

File: GsaGH/Components/0_Model/GetGeometry.cs [hunk 4]
```diff
@@ -1,92 +1,231 @@
...
-    /// <summary>
-    /// Component to retrieve geometric objects from a GSA model
-    /// </summary>
-    public class GetGeometry : GH_OasysTaskCapableComponent<GetGeometry.SolveResults>, IGH_PreviewObject, IGH_VariableParameterComponent
-  {
+namespace GsaGH.Components {
+  /// <summary>
+  ///   Component to retrieve geometric objects from a GSA model
+  /// </summary>
+  public class GetGeometry : GH_OasysTaskCapableComponent<GetGeometry.SolveResults>,
```
Comment (tlmnrnhrdt) [inline GsaGH/Components/0_Model/GetGeometry.cs:37]: why static methods before fields/properties?

File: GsaGH/Components/0_Model/GetGeometry.cs [hunk 5]
```diff
@@ -1,92 +1,231 @@
...
-    {
-    }
+
+    public GetGeometry() : base("Get Model Geometry",
+      "GetGeo",
+      "Get nodes, elements and members from GSA model",
+      CategoryName.Name(),
+      SubCategoryName.Cat0()) { }
+
     public override GH_Exposure Exposure => GH_Exposure.secondary;
```
Comment (tlmnrnhrdt) [inline GsaGH/Components/0_Model/GetGeometry.cs:132]: Properties should come before the constructor no?

File: GsaGH/Parameters/1_Properties/GsaProp2d.cs [hunk 6]
```diff
@@ -4,289 +4,235 @@
...
   /// <summary>
   /// Prop2d class, this class defines the basic properties and methods for any <see cref="GsaAPI.Prop2D"/>
   /// </summary>
-  public class GsaProp2d
-  {
+  public class GsaProp2d {
     #region fields
-    private int _id = 0;
+    private int _id;
     private Guid _guid = Guid.NewGuid();
```
Comment (tlmnrnhrdt) [inline GsaGH/Parameters/1_Properties/GsaProp2d.cs:19]: should not the public stuff be on top?

File: GsaGH/Parameters/1_Properties/GsaProp2d.cs [hunk 7]
```diff
@@ -4,289 +4,235 @@
...
     private Plane _localAxis = Plane.Unset;
     #endregion
 
     #region properties
-    internal Prop2D API_Prop2d
-    {
-      get
-      {
-        return this._prop2d;
+    internal Prop2D ApiProp2d {
```
Comment (tlmnrnhrdt) [inline GsaGH/Parameters/1_Properties/GsaProp2d.cs:26]: internal properties should be _camelCase?

File: GsaGH/Components/0_Model/GetGeometry.cs (hunk 4 above)
Comment (DominikaLos) [reply GsaGH/Components/0_Model/GetGeometry.cs:37]: will change the orger after fixing bugs

File: GsaGH/Parameters/1_Properties/GsaProp2d.cs (hunk 7 above)
Comment (DominikaLos) [reply GsaGH/Parameters/1_Properties/GsaProp2d.cs:26]: but what when we already have private field?

File: GsaGH/Parameters/1_Properties/GsaProp2d.cs (hunk 6 above)
Comment (DominikaLos) [reply GsaGH/Parameters/1_Properties/GsaProp2d.cs:19]: yup should be, but for this task I'm doing only renaming

File: GsaGH/Components/0_Model/GetGeometry.cs (hunk 5 above)
Comment (DominikaLos) [reply GsaGH/Components/0_Model/GetGeometry.cs:132]: yup should be, but for this task I'm doing only renaming

File: .editorconfig (hunk 3 above)
Comment (DominikaLos) [reply .editorconfig:95]: readability :) wouls like to leave s_ prefix

**Comment by tlmnrnhrdt** [comment]
Doing this here now: https://github.com/arup-group/GSA-Grasshopper/pull/352

File: GsaGH/Parameters/1_Properties/GsaProp2d.cs (hunk 7 above)
Comment (tlmnrnhrdt) [reply GsaGH/Parameters/1_Properties/GsaProp2d.cs:26]: yea, good point.. Did a quick google search but didn´t get an answer.
But I also think we should try and stick to the google style guide in this case somehow!

Comment (DominikaLos) [reply GsaGH/Parameters/1_Properties/GsaProp2d.cs:26]: can we leave it for now? and do it in other ticket then?

Comment (DominikaLos) [reply GsaGH/Parameters/1_Properties/GsaProp2d.cs:26]: image
everything inside class is private by default, and in this example (from google style guide) prop is in PascalCase

**Review by tlmnrnhrdt — APPROVED** [review]

**Pull request merged**